### PR TITLE
Add Support Gesture for A13

### PR DESCRIPTION
**Pull request by MrSluffy**
## Description

* Add Support Gesture for A13 without dropping A12/A12L
* Add support Icon (TaskIconCache)

Bug :
* PIP
* Split Screen

Added : ISplitScreenExt for A13 there's a better way to hook onGoingToRecentsLegacy but since I'm lazy I'll leave it to you guys :) Maybe you can clean it up :)

### CODE REVIEW

**Comment by azamuku** [comment]
Also classic crash when you interact with widgets
https://paste.evolution-x.org/pjtcBR

**Comment by wolf-yuan-6115** [comment]
What is this PR for 🤔
I'm not maintainer, I just saw this PR and it looks a little bit crazy...

Huge changes in single commit, no further information for this pull request and probably introduced more issues.

File: SystemUIShared/build.gradle [hunk 1]
```diff
@@ -17,6 +17,7 @@ android {
 }
 
 dependencies {
+    implementation project(path: ':hidden-api')
```
Comment (Goooler) [inline SystemUIShared/build.gradle]:
```suggestion
    implementation projects.hiddenApi
```

File: SystemUIShared/src/com/android/systemui/shared/system/ActivityManagerWrapper.java [hunk 2]
```diff
@@ -208,7 +210,8 @@ public void onAnimationStart(IRecentsAnimationController controller,
                         final RemoteAnimationTargetCompat[] wallpapersCompat =
                                 RemoteAnimationTargetCompat.wrap(wallpapers);
                         animationHandler.onAnimationStart(controllerCompat, appsCompat,
-                                wallpapersCompat, homeContentInsets, minimizedHomeBounds);
+                            wallpapersCompat, homeContentInsets, minimizedHomeBounds);
+
```
Comment (Goooler) [inline SystemUIShared/src/com/android/systemui/shared/system/ActivityManagerWrapper.java]:
```suggestion
```

File: SystemUIShared/src/com/android/systemui/shared/system/PackageManagerWrapper.java [hunk 3]
```diff
@@ -16,13 +16,15 @@
...
 package com.android.systemui.shared.system;
 
+import android.app.ActivityThread;
 import android.app.AppGlobals;
+import android.app.ApplicationPackageManager;
 import android.content.ComponentName;
 import android.content.Intent;
 import android.content.pm.ActivityInfo;
 import android.content.pm.IPackageManager;
+//import android.content.pm.IPackageManagerExt;
```
Comment (Goooler) [inline SystemUIShared/src/com/android/systemui/shared/system/PackageManagerWrapper.java]: ?

File: SystemUIShared/src/com/android/systemui/shared/system/PackageManagerWrapper.java [hunk 4]
```diff
@@ -73,12 +75,12 @@ public ComponentName getHomeActivities(List<ResolveInfo> allHomeCandidates) {
     /**
      * Determine the best Activity to perform for a given Intent.
      */
-    public ResolveInfo resolveActivity(Intent intent, @ResolveInfoFlags int flags) {
+    public ResolveInfo resolveActivity(Intent intent, int flags) {
         final String resolvedType =
-                intent.resolveTypeIfNeeded(AppGlobals.getInitialApplication().getContentResolver());
+            intent.resolveTypeIfNeeded(AppGlobals.getInitialApplication().getContentResolver());
```
Comment (Goooler) [inline SystemUIShared/src/com/android/systemui/shared/system/PackageManagerWrapper.java:78]: Do not leave unrelated changes here.

File: SystemUIShared/src/com/android/systemui/shared/system/PackageManagerWrapper.java [hunk 5]
```diff
@@ -73,12 +75,12 @@ public ComponentName getHomeActivities(List<ResolveInfo> allHomeCandidates) {
...
         try {
             return mIPackageManager.resolveIntent(
-                    intent, resolvedType, flags, UserHandle.getCallingUserId());
+                intent, resolvedType, flags, UserHandle.getCallingUserId());
```
Comment (Goooler) [inline SystemUIShared/src/com/android/systemui/shared/system/PackageManagerWrapper.java:81]: Same as above.

**Review by Goooler — CHANGES_REQUESTED** [review]

**Comment by Goooler** [comment]
You should revert all unrelated format changes.

**Comment by MrSluffy** [comment]
Will pr new, include fix for pip and splitscreen. will close this : update Move now : https://github.com/LawnchairLauncher/lawnchair/pull/3163

**Pull request closed without merging**